compiler/rustc_span/src/hygiene.rs

```diff
@@ -1473,29 +1415,10 @@ pub fn decode_syntax_context<D: Decoder, F: FnOnce(&mut D, u32) -> SyntaxContext
             Some(&ctxt) => ctxt,
```

Comment:
This `syntax_context_map` lookup is no longer necessary, `alloc_ctxt`/`apply_mark_internal` already performs it.